circus-train-aws/src/main/java/com/hotels/bdp/circustrain/aws/HadoopAWSCredentialProviderChain.java

```diff
@@ -42,7 +42,9 @@ public HadoopAWSCredentialProviderChain(String credentialProviderPath) {
   }
 
   public HadoopAWSCredentialProviderChain(Configuration conf) {
-    super(new JceksAWSCredentialProvider(conf), new EC2ContainerCredentialsProviderWrapper());
+    super(new JceksAWSCredentialProvider(conf),
```

Comment:
What's the significance of having the new provider in the `middle` of this chain (as opposed to the end)?

Reply:
it checks them in order, so it will have EC2 container credentials and it will try to use that instead. First one it finds wins

Reply:
Yeah, so this changes existing behaviour by putting something higher up in the chain, shouldn't it be added at the end instead?

Reply:
no because then always instance profile will be used we'll have no way to override it.
The AssumeRoleCP requires a property to be set otherwise it won't work similar to jceksCP. So I think this is the correct spot. 
Users had jckes configured which will still take priority.
Users relied on instanceProfile won't have the new property so that's ok.
New use case set AssumreRole prop and it will override InstanceProfile.

Reply:
OK, that makes sense - do we need to document the order somewhere? Also, would it be possible to have a test that asserts the ordering so someone in future doesn't make a mistake and change the order?

Reply:
again test not really possible the list of providers isn't exposed. I'll have a look at the readme see if we can document the order

Reply:
Yeah, OK, I see what you're saying, maybe just a comment here then explaining that order is significant.

Reply:
I tried last night to create a test for ordering of the chain, but it is a bit difficult since I couldn't figure out a way to mock the `STSAssumeRoleSessionCredentialsProvider` since it is not injected into the wrapper class.  Although maybe something with `spy` would work...

Reply:
@patduin - thank you for making it lazy fail.  We had it that way originally, I refactored last night when I was trying to write unit tests, then went to bed and immediately realized what I had done.  Bugged me all night and I couldn't sleep. :)